staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types_jsonschema.go

```diff
@@ -90,6 +90,29 @@ type JSONSchemaProps struct {
 	//      - type: string
 	//    - ... zero or more
 	XIntOrString bool `json:"x-kubernetes-int-or-string,omitempty" protobuf:"bytes,40,opt,name=xKubernetesIntOrString"`
+
+	// x-kubernetes-list-map-keys annotates lists with the x-kubernetes-list-type `map` by specifying the keys used
+	// as the index of the map.
+	//
+	// This tag MUST only be used on lists that have the "x-kubernetes-list-type"
+	// extension set to "map". Also, the values specified for this attribute must
+	// be a scalar typed field of the child structure (no nesting is supported).
```

Comment:
These keys are defaulted to null/"" right now. In the future we plan to also support defaulting for some of these keys (like the containerPort / protocol example from the built in apis)

I'll add validation for these keys having scalar type values

Reply:
>> how do we handle computing the key for an object if one of these attributes is not defined for it? do we default to null/""
>
> These keys are defaulted to null/"" right now

What happens if there's a default value for a field (e.g. `protocol` defaulting to "TCP") and the user doesn't specify it? Does the merge key get computed with `protocol:""` or `protocol:"TCP"`? We need a test to verify SSA interacts with CRD defaults the same way, since the handling chain is so different.